connection.go

```diff
@@ -72,11 +72,17 @@ Loop:
 				timer.Reset(i.timeout)
 			}
 		case <-expired:
-			for _, stream := range i.conn.streams {
-				stream.Reset()
-			}
+			i.conn.streamCond.L.Lock()
+			streams := i.conn.streams
+			i.conn.streams = make(map[spdy.StreamId]*Stream)
+			i.conn.streamCond.Broadcast()
+			i.conn.streamCond.L.Unlock()
+			go func() {
+				for _, stream := range streams {
+					stream.resetStream()
+				}
+			}()
 			i.conn.Close()
```

Comment:
Suggesting i.conn.Close is inside the goroutine before the resets? I don't like that hanging is a possibility on Close but I see it is only because the resetChan cannot be read from while this call is done. I will make that update, my main concern was i.conn.streams was updated before Close.